src/CollectionTracker.tsx

```diff
@@ -48,6 +48,113 @@ export interface CollectionTrackerArgs {
+ * content other than Navigate, then use the function returned from this
+ * hook to indicate the type and name of the action.
+ *
+ * Use of this hook is *optional*. If your list items don't contain controls
+ * for other action types, you don't need to call this. CollectionTracker
+ * will automatically log Navigate actions for the items in your list.
+ *
+ * List components wrapped with CollectionTracker will pass `setActionState`
+ * as a parameter to `renderItem`. You can use that parameter as a function
+ * to set the action type/name instead. This is mostly for use in class
+ * components. See `CollectionTracker`.
+ *
+ * # Usage
+ *
+ * ## Functional Components
+ *
+ * Suppose you have a list item with a "Like" button and a "More Like This"
+ * button, and tapping on these buttons should record `ActionType.Like` and
+ * `ActionType.Custom` respectively. In your `renderItem` function, add the
+ * following to the event handlers for your "Like" and "More Like This" buttons:
+ * ```
+ * const renderItem = ({ item }) => {
+ *   const setActionState = useCollectionActionState()
+ *   const likeButtonHandler = () => {
+ *     // Report this tap is ActionType.Like to Promoted.
+ *     // The rest of the details are automatically filled in.
+ *     setActionState({
+ *       actionType: ActionType.Like
+ *     })
+ *     likeItem(item)
+ *   }
+ *   const moreLikeThisButtonHandler = () => {
+ *     // Report a custom action to Promoted.
+ *     setActionState({
+ *       actionType: ActionType.Custom,
+ *       name: 'MoreLikeThis'
+ *     })
+ *     showMoreLikeThis(item)
+ *   }
+ *   const showButtonHandler = () => {
+ *     // Don't call setActionState to default Promoted to
+ *     // ActionType.Navigate.
+ *     showItem(item)
+ *   }
+ *   // ... Rest of handlers
+ *   return (
+ *     <MyItem>
+ *       // ... Rest of component
+ *       <ShowButton onPress={showButtonHandler} />
+ *       <LikeButton onPress={likeButtonHandler} />
+ *       <MoreLikeThisButton onPress={moreLikeThisButtonHandler} />
+ *     </MyItem>
+ *   )
+ * }
+ * ```
+ * This hook returns a function that you can call from your handler like
+ * the above example. If you don't call the function, then any taps on
+ * your list item will be recorded as `ActionType.Navigate`.
+ *
+ * To prevent Promoted from logging any actions, call
+ * `setActionState({ actionType: null })`.
+ *
+ * ## Class Components
+ *
+ * See `CollectionTracker` for details on accessory view action logging with
+ * class components.
+ *
+ * # Implementation Details
+ *
+ * Uses a React.Context passed in from the containing CollectionTracker-
+ * wrapped component. Your list items must be descendants of that component.
+ * Calling this hook outside of a functional component body or from a
+ * component that is not a CollectionTracker descendant will cause a runtime
+ * error.
+ *
+ * The function returned from this hook must be called synchronously in your
```

Comment:
Semi-related to this, what's the failure pattern if people use CollectionTracker incorrectly?

I'm also curious if we can eventually get signals to report back to their devs.

Reply:
Eventually, I'll add more usage guides to the SDK to cover cases like these. That will get in after the Hipcamp CollectionTracker integration is complete. In this specific case, calling this function outside of its intended use case will do nothing, but we can detect this in the future and throw exceptions in dev/log in prod.